numpy/_core/src/multiarray/stringdtype/static_string.c

```diff
@@ -685,9 +685,6 @@ NpyString_dup(const npy_packed_static_string *in,
     _npy_static_string_u *in_u = (_npy_static_string_u *)in;
     char *in_buf = NULL;
     npy_string_arena *arena = &in_allocator->arena;
-    if (arena->buffer == NULL) {
-        return -1;
-    }
```

Comment:
this check is unnecessary, all the code below it can run even if `arena->buffer` is `NULL`